apps/web/pages/api/integrations.ts

```diff
@@ -39,12 +49,84 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       data: {
         credentials: {
           delete: {
-            id,
+            id: credentialToDeleteId,
           },
         },
       },
     });
 
+    /** This validates if you disconnect the Stripe credential.
+    When you disconnect this credential, you have to delete the information from payment 
+    and then update the booking status to Rejected, this is just for those booking that are unconfirmed and 
+    unpaid*/
+    if (paymentCredential?.id === credentialToDeleteId) {
+      try {
+        const bookingWithPaymentIds = await prisma.booking
+          .findMany({
+            where: {
+              userId: session?.user?.id,
+              paid: false,
+              NOT: {
+                payment: {
+                  every: {
+                    booking: null,
+                  },
+                },
+              },
+            },
+            select: {
+              id: true,
+            },
+          })
+          .then((bookings) => bookings.map((booking) => booking.id));
+        const deletePayments = prisma.payment.deleteMany({
+          where: {
+            bookingId: {
+              in: bookingWithPaymentIds,
+            },
+            success: false,
+          },
+        });
+
+        const updateBookings = prisma.booking.updateMany({
+          where: {
+            id: {
+              in: bookingWithPaymentIds,
+            },
+          },
+          data: {
+            rejected: true,
+            status: "REJECTED",
```

Comment:
I think Rejected means the owner has explicitly rejected to meet with someone else, on this case it can be Cancelled and use cancellationReason = "Payment provider got removed".
Probably we need to revise what we wanna do here with bookings that had payment option but provider got removed.
I would suggest we tell the user what is gonna happen when disconnecting from stripe and has some events that rely on it, whether it's cancelling them or just remove they payment required fields.
cc @zomars

Reply:
Agreed

Reply:
Agreed

Reply:
Working on 👍